test/printf.c

```diff
@@ -0,0 +1,20 @@
+#include <stdio.h>
```

Comment:
Hmm, asm2wasm does emit `i64.store` in printf code, but looks like of only 8 bytes. So yeah, I guess the test should go in the `wasm_backend` section.

That raises another question, when does the wasm backend emit `i64.store4` etc.? Thinking about it now, it seems surprising it's much useful. At best it can save a little by replacing `i32.store + i32.wrap`? And looking in asm2wasm output, I don't see cases of that which could be folded, so I am wondering what the wasm backend is doing differently?